### PR TITLE
Support building with gdmd host compiler

### PR DESCRIPTION
**Pull request by MoritzMaxeiner**
Depends on [this](https://github.com/D-Programming-GDC/gdmd/pull/6) gdmd pull request.

### CODE REVIEW

File: cmake/Modules/ExtractDMDSystemLinker.cmake [hunk 1]
```diff
@@ -34,6 +34,8 @@ execute_process(
     RESULT_VARIABLE result_code
     OUTPUT_VARIABLE stdout
     OUTPUT_STRIP_TRAILING_WHITESPACE
+    ERROR_VARIABLE stdout
+    ERROR_STRIP_TRAILING_WHITESPACE
```
Comment (kinke) [inline cmake/Modules/ExtractDMDSystemLinker.cmake]: Thanks! - Merging stderr output into the `stdout` variable breaks the linker line detection if the linker writes to stderr, as shown by Shippable CI for AArch64.

Comment (MoritzMaxeiner) [reply cmake/Modules/ExtractDMDSystemLinker.cmake]: Okay, that's concerning, but that's what CI is for. gdmd requires parsing `stderr`, so I'll take this into account.

Comment (MoritzMaxeiner) [reply cmake/Modules/ExtractDMDSystemLinker.cmake]: Should be solved in the new commit?

**Comment by MoritzMaxeiner** [comment]
@kinke semaphore ci job was killed while executing. Should I force push again to trigger a new job?

**Comment by kinke** [comment]
No worries, that happens quite frequently lately, unfortunately.

**Comment by MoritzMaxeiner** [comment]
The required [gdmd PR](https://github.com/D-Programming-GDC/gdmd/pull/6) has been merged.

**Comment by kinke** [comment]
I'd like to kill the detection of `ldc2` and `gdc`, `D_COMPILER_DMD_COMPAT` etc., as it is currently in vain and will most likely remain unsupported in the foreseeable future. Any objections before I get to it?

File: cmake/Modules/ExtractDMDSystemLinker.cmake (hunk 1 above)
Comment (kinke) [reply cmake/Modules/ExtractDMDSystemLinker.cmake]: Only as long as the linker used by gdc doesn't produce any warnings I guess, but oh well. ;)

**Comment by MoritzMaxeiner** [comment]
@kinke I'm sorry, I don't think I understand what exactly you're asking. Do you want to close this PR without merging?

**Comment by kinke** [comment]
Ah sry, that wasn't directed at you, but at other LDC devs.

**Comment by JohanEngelen** [comment]
> I'd like to kill the detection of `ldc2` and `gdc`, `D_COMPILER_DMD_COMPAT` etc., as it is currently in vain and will most likely remain unsupported in the foreseeable future. Any objections before I get to it?

I think that's ok. It'd be nice to change the error `No D compiler found` to something like `No D compiler found (dmd, ldmd2, gdmd)`, so it is more clear that it is looking for DMD-compatible compiler commandline interface (but writing "DMD compatible" won't make sense to most people).

**Pull request merged**